docs/recipes/plugin/lighthouse-plugin-preload-as/preload-as-audit.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('lighthouse').Audit;
+
+/**
+ * @fileoverview A fake additional check of the robots.txt file.
+ */
+
+// https://fetch.spec.whatwg.org/#concept-request-destination
+const allowedTypes = new Set(['font', 'image', 'script', 'serviceworker', 'style', 'worker']);
+
+class LoadAudit extends Audit {
+  static get meta() {
+    return {
+      id: 'preload-as-audit',
```

Comment:
seems weird to put 'audit' in the id. how about 'preload-as' as the id and `preload-as.audit.js` as filename?

Reply:
> seems weird to put 'audit' in the id. how about 'preload-as' as the id and `preload-as.audit.js` as filename?

I'd agree with you if this was one of our audits, but it's helpful to have it in the filename, and it seems confusing to me to not have it match the id at that point

Reply:
> everybody wins

that's the opposite of what I said

> or move the audit js file into `/audits/preload-as.js`

adding a level of indirection seems worse for an example

Reply:
I'm definitely in the _preload-as-audit_ is gross for including "audit".  I personally like having a directory structure, to me it kind of keeps things all in their place.  I like `audits/preload-as.js`.